includes/classes/PluginSupport/ScriptedInstallHelpers.php

```diff
@@ -0,0 +1,234 @@
+<?php
+/**
+ * @copyright Copyright 2003-2024 Zen Cart Development Team
+ * @license http://www.zen-cart.com/license/2_0.txt GNU Public License V2.0
+ * @version $Id:      $
+ *
+ */
+
+namespace Zencart\PluginSupport;
+
+use queryFactory;
+
+trait ScriptedInstallHelpers
+{
+    protected queryFactory $dbConn;
+
+    /**
+     * Get details of current configuration record entry, false if not found.
+     * Optional: when $only_check_existence is true, will simply return true/false.
+     */
+    public function getConfigurationKeyDetails(string $key_name, bool $only_check_existence = false): array|bool
+    {
+        $sql = "SELECT * FROM " . TABLE_CONFIGURATION . " WHERE configuration_key = '" . zen_db_input($key_name) . "'";
+        $result = $this->dbConn->Execute($sql, 1);
+
+        // false if not found, or if existence-check fails
+        if ($only_check_existence || $result->EOF) {
+            return !$result->EOF;
+        }
+
+        return $result->fields;
+    }
+
+    public function addConfigurationKey(string $key_name, array $properties): int
+    {
+        $exists = $this->getConfigurationKeyDetails($key_name, true);
+        if ($exists) {
+            return 0;
```

Comment:
Shouldn't this return the `configuration_id` if the configuration-key exists (i.e. `$exists` is an array)?

Reply:
`exists` will only be a `bool` in this case because of the 2nd param sent on line 36.
My thinking was that if it already exists, there won't be an insert, so returning 0 is an easy indicator of that (and `INSERT IGNORE` wouldn't return the id either).

But yes we could change it up and return the found ID without reporting any clue that it already existed.